datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -83,6 +83,9 @@ impl CoalesceBatchesExec {
             input.execution_mode(),                 // Execution Mode
         )
     }
+    pub fn set_target_batch_size(&mut self, siz: usize) {
+        self.target_batch_size = siz;
```

Comment:
Instead of overwriting `target_batch_size`. We can add `fetch: Option<usize>`. `CoalesceBatchesExec` can emit when hit to this count also. As well as `target_batch_size`.

Reply:
sure